### DISCUSSION TITLE
Multiple measurement units in one ingredient

### DISCUSSION THREAD

**Discussion by TheHYPO35** [mealie-recipes/mealie#3127] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I have an occasional recipe that mixes units. For example "1 cup plus 1 tbsp sugar". Right now, the system only allows a single measurement unit per ingredient.

Current workaround options are :

1.  to convert it all into a single messy unit (1.0625 cups)
2.  make two entries for the same ingredient and add a note that they are combined
3.  don't use the ingredient quantity system for that ingredient and put the whole entry into the 'note' ("1 cup plus 1 tbsp sugar")
4.  don't use the ingredient quantity system at all

### Please provide a concise description of the feature that would resolve your issue.

It would be nice to be able to have + sign where you could add a second measurement unit to the same entry. 

Alternatively, make two different entries for the same ingredient (one for each unit) but have a button that links the lines and displays them as a single entry.

### Please consider and list out some caveats or tradeoffs made in your design decision

For the second alternative proposed, any linking of two lines should be manual, as there are other cases where a recipe intentionally has two separate entries for the same ingredient (for two different steps or components, for example).

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by ModestTG** [comment]
This would also be helpful if you wanted to enter recipes for both weight and volume units. I like to bake by weight, but most recipes are written for volumetric measurements. So having both units on one ingredient would be helpful, especially for sharing recipes. For example:

- 120g (1 C) all-purpose flour
Or
- 1 C (120g) all-purpose flour

**Comment by crimsoncor** [comment]
See my reply here which I think might solve the issue raised in the initial request.

https://github.com/mealie-recipes/mealie/discussions/3881#discussioncomment-14084487

So in your example, you'd enter the recipe amount as 17 tbsp and then the unit conversion would turn that into 1 cup + 1 tbsp. 

the reply from @ModestTG seems like it should be a different ticket, because that is looking at having 2 equivalent measures for the same ingredient. That would also be very useful but is less about conversions and more about adding additional complexity to recipe entries.

**Reply by TheHYPO35** [reply]
While that would function, it would be more of a hassle to have to manually convert when entering a recipe from 1 cup + 1 tbsp to look up how many tbsp are in a cup and combining the two measures (assuming they even divide evenly into each other).

This is not a comment against the conversion suggestion in the other thread, but I don't think it's the best solution for this particular problem (which I think is still just having the ability to add a second (or more) measurement field to an ingredient.

I don't necessarily agree that @ModestTG's valid use case is necessarily a separate ticket, as the ability to add a second measurement to an ingredient could easily include an "and"/"or" option - so it could display as "1 cup + 1 tbsp", or  "1 cup (120g)" /"1 cup or 120g" or some other display to show alternates.

Here in Canada, we certainly run into cases where we get recipes with American measure (an 8oz can of something) and having the 'or # grams / ml' would be useful without wanting to removing the original recipe's measure

**Reply by crimsoncor** [reply]
I think we disagree on the hassle level. I'd have no problems doing the conversion once when inputting the recipe if the trade-off is that mealie will correctly handle scaling for me. For example, if I do `1 tbsp + 1 tsp` and then scale 3x, I'd want that to show as `4 tbsp` not `3 tbsp + 3 tsp`, etc, etc. 

Those things, I guess, are not mutually exclusive, but it feels to me like if a user were to enter something as two different units, that automatically collapsing it might not be the desired behavior (for example if the user for some reason entered `3 tbsp + 3 tsp`, mealie probably shouldn't collapse that down to `4 tbsp`). But those are all things that can be controlled with prefs as well, at the expense of more user complexity. 

Looking at the mealie code, one advantage to my suggestion is that it doesn't require any changes to the existing recipe schema, making implementation much easier.
